src/model/implementation/object-type-base.ts

```diff
@@ -22,6 +23,7 @@ export abstract class ObjectTypeBase extends TypeBase {
             ...customFields
         ];
         this.fieldMap = new Map(this.fields.map((field): [string, Field] => [ field.name, field ]));
+        this.namespacePath = input.namespacePath || [];
```

Comment:
This should be in `TypeBase`, there is nothing special about object types with regards to namespaces.